src/Extractor/HttpClient.php

```diff
@@ -100,11 +88,12 @@ public function setLogger(LoggerInterface $logger)
      *     - content type header.
      *
      * @param string $url
+     * @param array  $httpHeader Custom HTTP Headers from SiteConfig
      * @param bool   $skipTypeVerification Avoid mime detection which means, force GET instead of potential HEAD
      *
      * @return array With keys effective_url, body & headers
      */
-    public function fetch($url, $skipTypeVerification = false)
+    public function fetch($url, $httpHeader = array(), $skipTypeVerification = false)
```

Comment:
Could you move that new parameter as third one? If we inject it in the second arg, it'll require to release a new major version because the signature of this public method will change. I know, it'll require to put the `false` argument for `$skipTypeVerification` every where you'll have to inject `$httpHeader` but it'll avoid BC